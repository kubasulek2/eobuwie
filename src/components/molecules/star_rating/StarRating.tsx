import React, {VFC} from 'react';
import {scoreToStarValue} from "../../../utils/scoreToStarValue";
import Star from "../../atoms/star/Star";
import {StarRatingProps} from "./types";
import styles from "./StarRating.module.css";
import {joinClassNames} from "../../../utils/joinClassNames";
import {text_medium_dark} from "../../../styles/colors";

/**
 * Component displays 5 star rating.
 */
const StarRating: VFC<StarRatingProps> = ({score, votes, id, size = "medium"}) => {
  return (
    <div
      data-testid="star_rating"
      className={joinClassNames(styles.container, styles["container_" + size])}
      id={id}
    >
      <div
        className={styles.wrapper}
        aria-label={`Rating: ${ score }. Based on ${ votes } votes.`}
      >
        {[1, 2, 3, 4, 5].map(num => <Star rating={scoreToStarValue(score, num)} size={size} key={num} />)}
      </div>
      <span className={joinClassNames(styles.votes, text_medium_dark, styles["votes_" + size])}>{votes}</span>
    </div>
  );
};

export default StarRating;
