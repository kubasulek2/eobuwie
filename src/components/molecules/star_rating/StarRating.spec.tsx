import { render, screen } from '@testing-library/react';
import StarRating from "./StarRating";

describe('snapshot', () => {
	it('Matches the snapshot', () => {
		expect(render(<StarRating score={5} votes={100} />)).toMatchSnapshot();
	});
});



it('Renders star rating', () => {
  render(<StarRating score={5} votes={100} />);
  expect(screen.getByTestId('star_rating')).toBeInTheDocument();
});


it('Has proper label', () => {
  render(<StarRating score={5} votes={100} />);
  screen.getByLabelText('5', {exact: false}); // 5 is rating, and must be in label text 
  screen.getByLabelText('100', {exact: false}); // 5 is votes number, and must be in label text 
});

it('Displays right score', () => {
  render(<StarRating score={3.5} votes={100} />);
  expect(screen.getAllByText('star')).toHaveLength(3); // 4 full stars
  expect(screen.getAllByText('star_half')).toHaveLength(1); // 1 half star
  expect(screen.getAllByText('star_border')).toHaveLength(1); // 1 empty star
});

it('Displays right vote\'s number', () => {
  render(<StarRating score={4.5} votes={100} />);
  screen.getByText("100");
});


