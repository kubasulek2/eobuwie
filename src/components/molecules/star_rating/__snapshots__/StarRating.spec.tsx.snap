// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container container_medium"
        data-testid="star_rating"
      >
        <div
          aria-label="Ocena: 5. Na podstawie 100 głosów."
          class="wrapper"
        >
          <div
            data-testid="star"
            role="presentation"
          >
            <span
              class="material-icons teal medium"
            >
              star
            </span>
          </div>
          <div
            data-testid="star"
            role="presentation"
          >
            <span
              class="material-icons teal medium"
            >
              star
            </span>
          </div>
          <div
            data-testid="star"
            role="presentation"
          >
            <span
              class="material-icons teal medium"
            >
              star
            </span>
          </div>
          <div
            data-testid="star"
            role="presentation"
          >
            <span
              class="material-icons teal medium"
            >
              star
            </span>
          </div>
          <div
            data-testid="star"
            role="presentation"
          >
            <span
              class="material-icons teal medium"
            >
              star
            </span>
          </div>
        </div>
        <span
          class="votes text_medium votes_medium"
        >
          100
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container container_medium"
      data-testid="star_rating"
    >
      <div
        aria-label="Ocena: 5. Na podstawie 100 głosów."
        class="wrapper"
      >
        <div
          data-testid="star"
          role="presentation"
        >
          <span
            class="material-icons teal medium"
          >
            star
          </span>
        </div>
        <div
          data-testid="star"
          role="presentation"
        >
          <span
            class="material-icons teal medium"
          >
            star
          </span>
        </div>
        <div
          data-testid="star"
          role="presentation"
        >
          <span
            class="material-icons teal medium"
          >
            star
          </span>
        </div>
        <div
          data-testid="star"
          role="presentation"
        >
          <span
            class="material-icons teal medium"
          >
            star
          </span>
        </div>
        <div
          data-testid="star"
          role="presentation"
        >
          <span
            class="material-icons teal medium"
          >
            star
          </span>
        </div>
      </div>
      <span
        class="votes text_medium votes_medium"
      >
        100
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
